### PR TITLE
Fix Xcode warnings for constants, downcasting and dead code

### PR DESCRIPTION
**Pull request by pjorquera**
* Use `let` instead var for constants when possible.
* Fixed downcasting to unrelated types warnings.
* Fixed dead code warnings using `#if` `#else` `#endif` when possible.

### CODE REVIEW

File: RadarCovid/Data/WS/Models.swift [hunk 1]
```diff
@@ -31,9 +31,9 @@ open class Response<T> {
     }
 
     public convenience init(response: HTTPURLResponse, body: T?) {
-        let rawHeader = response.allHeaderFields
+        let rawHeader = response.allHeaderFields as! [String: String]
```
Comment (miguelangel-dev) [inline RadarCovid/Data/WS/Models.swift]: It is an insecure conversion from `[AnyHashable: Any]` to `[String: String]` original code was fine.

Not sure about why this change, but if Xcode is launching some warnings, you can try with:

```swift
for case let (key as String, value as String) in rawHeader
```

File: RadarCovid/Application/JailBreakDetect.swift [hunk 2]
```diff
@@ -24,7 +24,8 @@ class JailBreakDetect {
         #if arch(i386) || arch(x86_64)
             // This is a Simulator not an idevice
             return false
-        #endif
+
```
Comment (miguelangel-dev) [inline RadarCovid/Application/JailBreakDetect.swift]: as a suggestion, you can extract the macros to an external descriptive method:
```swift
static var isSimulator: Bool {
        #if arch(i386) || arch(x86_64)
            return true
        #else
            return false
        #endif
    }
```
On this way, you reduce 1-level of indentation and keep code more declarative. For example you can use guard: `guard !isSimulator else { return false }`

Comment (RobertoEstrada) [reply RadarCovid/Application/JailBreakDetect.swift]: The code should not rely on the current architecture to check if we're on the simulator or not because the simulator will run on ARM architecture on the future Apple Silicon macs and the current mac-catalyst platform is x86_64.

So, to check if we're on a simulator or not, the proper solution is `#if targetEnvironment(simulator)` wherever this condition is required. This conditionally builds simulator or actual device code and does not relegate the check until run time.

File: RadarCovid/Data/WS/Models.swift (hunk 1 above)
Comment (applecoding) [reply RadarCovid/Data/WS/Models.swift]: For the love of Jobs... ! is forbidden...
<pre>
var header = [String: String]()
for (key, value) in rawHeader {
      if let key = key as? String, let value = value as? String {
            header[key] = value
      }
}
</pre>

This is the way

File: RadarCovid/Application/JailBreakDetect.swift (hunk 2 above)
Comment (pjorquera) [reply RadarCovid/Application/JailBreakDetect.swift]: Fixed in 6e03017da35a4405c358e9388bb378b80e169a7d

File: RadarCovid/Data/WS/Models.swift (hunk 1 above)
Comment (pjorquera) [reply RadarCovid/Data/WS/Models.swift]: Was fixed in 02f6237e73b4930faf07056419f955999c1dda32

Comment (applecoding) [reply RadarCovid/Data/WS/Models.swift]: > Was fixed in [02f6237](https://github.com/RadarCOVID/radar-covid-ios/commit/02f6237e73b4930faf07056419f955999c1dda32)

Bad fixed... you cannot use `as!` because is a forced conversion. You must use always conditional.

Comment (pjorquera) [reply RadarCovid/Data/WS/Models.swift]: 02f6237 does not use `as!`, and I don't totally agree about using *always* optional casting...

Anyway, fixed (again) in b53c6b0a9de3f1d03391a9db80b63a095ffda3d7

**Comment by gmoraleda** [comment]
Please, read: https://github.com/RadarCOVID/radar-covid-ios/blob/master/CONTRIBUTING.md
